<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1853</width>
    <height>1164</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="MyOpenGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>0</y>
      <width>1341</width>
      <height>1101</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>471</width>
      <height>763</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_18">
     <item>
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="2" column="0">
        <widget class="QRadioButton" name="radioButton">
         <property name="styleSheet">
          <string notr="true">QRadioButton {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>Центральная проекция</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="autoRepeat">
          <bool>false</bool>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QRadioButton" name="radioButton_2">
         <property name="styleSheet">
          <string notr="true">QRadioButton {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>паралельная проекция</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="Name">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>Name_file</string>
         </property>
         <property name="scaledContents">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pB_resetParam">
         <property name="text">
          <string>Сбросить параметры</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="Open_file">
         <property name="styleSheet">
          <string notr="true">QPushBotton {
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 18px 1px 3px;
    min-width: 6em;
}</string>
         </property>
         <property name="text">
          <string>Выбрать файл</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="filename">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>File_path</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="styleSheet">
            <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
           </property>
           <property name="text">
            <string>Ox</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="SpinBox_moveX">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="styleSheet">
            <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
           </property>
           <property name="text">
            <string>Oy</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="SpinBox_moveY">
           <property name="minimum">
            <double>-100.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="styleSheet">
            <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
           </property>
           <property name="text">
            <string>Oz</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="SpinBox_moveZ">
           <property name="minimum">
            <double>-105.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer_15">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="SpinBox_Scale">
         <property name="maximumSize">
          <size>
           <width>70</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="decimals">
          <number>2</number>
         </property>
         <property name="maximum">
          <double>100.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="l_scale">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>Масштаб</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>2</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>1</number>
         </property>
         <item>
          <widget class="QLabel" name="l_rot_x">
           <property name="text">
            <string>X</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="SpinBox_X_rot">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="minimum">
            <number>-1</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDial" name="dial_X_rot">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>76</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>76</height>
            </size>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>359</number>
           </property>
           <property name="tracking">
            <bool>true</bool>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
           <property name="wrapping">
            <bool>true</bool>
           </property>
           <property name="notchTarget">
            <double>6.000000000000000</double>
           </property>
           <property name="notchesVisible">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="spacing">
          <number>1</number>
         </property>
         <item>
          <widget class="QLabel" name="l_rot_y">
           <property name="text">
            <string>Y</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="SpinBox_Y_rot">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="minimum">
            <number>-1</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDial" name="dial_Y_rot">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>76</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>76</height>
            </size>
           </property>
           <property name="maximum">
            <number>359</number>
           </property>
           <property name="tracking">
            <bool>true</bool>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
           <property name="wrapping">
            <bool>true</bool>
           </property>
           <property name="notchTarget">
            <double>6.000000000000000</double>
           </property>
           <property name="notchesVisible">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>1</number>
         </property>
         <item>
          <widget class="QLabel" name="l_rot_z">
           <property name="text">
            <string>Z</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="SpinBox_Z_rot">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="minimum">
            <number>-1</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDial" name="dial_Z_rot">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>76</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>76</height>
            </size>
           </property>
           <property name="maximum">
            <number>359</number>
           </property>
           <property name="tracking">
            <bool>true</bool>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
           <property name="wrapping">
            <bool>true</bool>
           </property>
           <property name="notchTarget">
            <double>6.000000000000000</double>
           </property>
           <property name="notchesVisible">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <widget class="QLabel" name="label_12">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>f :</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="num_of_vertices">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_11">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>v : </string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="num_of_facets">
         <property name="styleSheet">
          <string notr="true">QLabel {
	border: 2px solid blue;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    padding: 6px;
}</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="2" column="0">
        <widget class="QRadioButton" name="dash_line">
         <property name="text">
          <string>Пунктирная</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QRadioButton" name="vertex_disable">
         <property name="text">
          <string>Отсутствует</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Тип вершин</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QRadioButton" name="disable_line">
         <property name="text">
          <string>Отсутствует</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QRadioButton" name="circle">
         <property name="text">
          <string>Круг</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Тип линий:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="line">
         <property name="text">
          <string>Сплошная</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QRadioButton" name="square">
         <property name="text">
          <string>Квадрат</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="1" column="2">
        <widget class="QPushButton" name="color_background">
         <property name="text">
          <string>Фон</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="color_lines">
         <property name="text">
          <string>Линий</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="color_point">
         <property name="text">
          <string>Вершин</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="l_colorSettings">
         <property name="text">
          <string>Настройка цвета</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <spacer name="verticalSpacer_16">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QDoubleSpinBox" name="doubleSpinBox_8">
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>15.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="doubleSpinBox_9">
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>50.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Толщина линий:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Тощина точек:</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer_17">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="exit_program">
       <property name="styleSheet">
        <string notr="true">QPushButton {
	border: 2px solid red;
    border-radius: 4px;
    border-style: outset;
    border-width: 2px;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}
</string>
       </property>
       <property name="text">
        <string>Выйти из программы</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>820</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Скрин</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pB_GIF">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>830</y>
      <width>114</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Создание .gif</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1853</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyOpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">myopenglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
